ci-operator/step-registry/gcp/provision/mirror-dns/gcp-provision-mirror-dns-ref.yaml

```diff
@@ -0,0 +1,21 @@
+ref:
+  as: gcp-provision-mirror-dns
+  from_image:
+    namespace: ocp
+    name: "4.10"
+    tag: upi-installer
+  grace_period: 10m
+  commands: gcp-provision-mirror-dns-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  credentials:
```

Comment:
In the process, I think we do not need these credentials. 

Another suggestion, is it possible to merge `gcp-provision-mirror-dns-commands.sh` to `gcp/provision/bastionhost/gcp-provision-bastionhost-commands.sh`, introduce `REGISTER_MIRROR_REGISTRY_DNS` env to control it. Similar with `azure/provision/bastionhost/azure-provision-bastionhost-commands.sh`.

Some benifits:
1. only one ref to combine mirror registry configure and launch, also dns register.
2. reduce some commands, e.g: auth login, yq download, and so on.

Reply:
In fact, initially I'd ever put things into the bastion host step, e.g. mirror, proxy, etc., but you suggested me to remove them and leave only launching the bastion host that time. So?

Reply:
Sorry, I can not remember when I mentioned that. If agree, let us merge them together.